src/penn_chime/models.py

```diff
@@ -72,187 +77,181 @@ def __init__(self, p: Parameters):
-            p.n_days_since_first_hospitalized = int(self.argmin_ds(p))
-            p.date_first_hospitalized = date.today() - timedelta(days=p.n_days_since_first_hospitalized)
-            admits_df = build_admits_df(dispositions_df, p.n_days_since_first_hospitalized)
-            census_df = build_census_df(admits_df, lengths_of_stay, p.n_days_since_first_hospitalized)
-            self.admits_df = admits_df
-            self.census_df = census_df
-        if p.n_days_since_first_hospitalized is not None and p.doubling_time is None:
-            # optimize doubling_time
-            argmin_dt = None
+        if p.date_first_hospitalized is None and p.doubling_time is not None:
+            print('%s: use doubling_time.' % (datetime.now(),))
+            n_days_since = int(get_argmin_ds(census_df, p.current_hospitalized))
+
+            raw_df = sim_sir_df(
+                susceptible,
+                infected,
+                p.recovered,
+                beta,
+                gamma,
+                p.n_days + n_days_since,
+                -n_days_since,
+            )
+            dispositions_df = build_dispositions_df(raw_df, rates, p.market_share, p.current_date)
+            admits_df = build_admits_df(dispositions_df)
+            census_df = build_census_df(admits_df, lengths_of_stay)
+
+        elif p.date_first_hospitalized is not None and p.doubling_time is None:
+            print('%s: using date_first_hospitalized.' % (datetime.now(),))
             min_loss = 2.0**99
-            censes = dict()
-            current_infecteds = dict()
-            for dt in np.linspace(1,15,29):
-                censes[dt], current_infecteds[dt] = self.run_projection_dt(p, dt)
-                # self.current_infected = current_infecteds[dt] # log it into state for no reason really
-                self.census_df = censes[dt] # log it into state for loss
-                loss_dt = self.loss_dt(p)
-                if loss_dt < min_loss:
-                    min_loss = loss_dt
-                    argmin_dt = dt
-            self.census_df = censes[argmin_dt]
-            p.doubling_time = argmin_dt
-            current_infected = current_infecteds[argmin_dt]
-
-            infected = 1 / p.hospitalized.rate / p.market_share
+            dt = census_df = current_infected = None
+            for i_dt in np.linspace(1,15,29):
+                i_census_df, i_current_infected = get_census_and_infected_projection(
+                    rates=rates,
+                    lengths_of_stay=lengths_of_stay,
+                    n_days_since=n_days_since,
+                    doubling_time=i_dt,
+                    p=p)
+                loss = get_loss(i_census_df, p.current_hospitalized, n_days_since)
+                if loss < min_loss:
+                    min_loss = loss
+                    dt, census_df, current_infected = i_dt, i_census_df, i_current_infected
+            p.doubling_time = dt
+
```

Comment:
_correct_. factored out my dictionaries and use of state. this is how it should be. well done. 

i'm spending extra time verifying the equivalence